src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets

```diff
@@ -1,27 +1,17 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
 <Project>
   <UsingTask TaskName="Microsoft.DotNet.PackageValidation.ValidatePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
-  <UsingTask TaskName="Microsoft.DotNet.PackageValidation.GetLastStablePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
 
-  <PropertyGroup>
-    <!-- On static graph build, the CollectPackageDownloads target is invoked per target and there is no other way to run logic only once for all tfms.
-         Hence the GetLastStablePackage feature should not be used when using static graph build as it results in a remote call per tfm. -->
-    <AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets Condition="'$(RestoreUseStaticGraphEvaluation)' == 'true'">CollectPackageDownloads</AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets>
-    <!-- For a non static graph restore with a single tfm, hook onto CollectPackageDownloads which will then be invoked just once. -->
-    <AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets Condition="'$(AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets)' == '' and '$(TargetFrameworks)' == '' and '$(TargetFramework)' != ''">CollectPackageDownloads</AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets>
-    <!-- For a non static graph restore with multiple tfms, hook onto the target that fans out to the individual tfm restore graph walk operations. -->
-    <AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets Condition="'$(AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets)' == ''">_GenerateProjectRestoreGraphAllFrameworks</AddPackageValidationBaselinePackageAsPackageDownloadBeforeTargets>
-  </PropertyGroup>
+  <ItemGroup Condition="'$(PackageValidationBaselinePath)' == '' and '$(PackageValidationBaselineVersion)' != ''">
+    <PackageDownload Include="$(PackageId)" Version="[$(PackageValidationBaselineVersion)]" />
```

Comment:
I wonder if it's worth checking if the user for whatever reason added PackageDownload for that packageId already. Also, should we allow folks to specify a different packageID? For example, what happens if someone renamed ther package across releases, but still wants to perform validation to make sure their new package supports all of the previous frameworks and is api compatible?

Reply:
> Also, should we allow folks to specify a different packageID? For example, what happens if someone renamed ther package across releases, but still wants to perform validation to make sure their new package supports all of the previous frameworks and is api compatible?

They can use PackageValidationBaselinePath property to use against versions of different name.


> I wonder if it's worth checking if the user for whatever reason added PackageDownload for that packageId already.

I don't think we see any performance degradation or adding extra network calls. So i think we can avoid warning user about it.

Reply:
I think being able to specify a different PackageId when using PackageDownload does make sense and we already had that feature before. My suggestion would be to add that part of the code back that defines `PackageValidationBaselineName` and default it to PackageId.

Reply:
>They can use PackageValidationBaselinePath property to use against versions of different name.

Good point.

> I don't think we see any performance degradation or adding extra network calls. So i think we can avoid warning user about it.

I wasn't too concerned about that, but instead more about what happens if the PackageDownload specified by the user is a different version than the one we are using, will that still work?

Reply:
> I wasn't too concerned about that, but instead more about what happens if the PackageDownload specified by the user is a different version than the one we are using, will that still work?

Yes, PackageDownload is specifically designed for that scenario which is being able to restore different versions of the same package.

Reply:
> My suggestion would be to add that part of the code back that defines PackageValidationBaselineName and default it to PackageId

What are the cases where we want packageValidationBaselineName to be different from packageID ? For jose scenario where a user decides to change names between releases, they can directly use PackageValidationBaselinePath

Reply:
You can male the same argument for the baseline version number. The idea of providing a way to override the package name is just convenience, so user doesn’t have to do packagedownload themselves and construct the path when we can do that for them.

Reply:
> . The idea of providing a way to override the package name is just convenience, so user doesn’t have to do packagedownload themselves and construct the path when we can do that for them.

we already have the packageId for most of the cases where we want to override the packageName. A person can set packageId if they want.

Reply:
> where a user decides to change names between releases, they can directly use PackageValidationBaselinePath

How would that work in CI? Imagine we rename a package in dotnet/runtime but still would like to validate against the baseline? How would we accomplish that in CI then?

Reply:
> You can make the same argument for the baseline version number. 

Version is different in the sense we can have different versions that a user might want to test against.

Reply:
I think @ViktorHofer suggestion of allowing users to optionally override package baseline id is a good one we should take, but that is also a new feature which is not exactly tied to this PR so I'm fine with adding that later and wouldn't want to block this PR on that discussion so I'm fine with the way this code is for now, and we should just consider adding the package id override in a subsequent PR.